fe/fe-core/src/main/java/com/starrocks/statistic/StatisticsCollectJobFactory.java

```diff
@@ -115,32 +121,63 @@ private static void createJob(List<StatisticsCollectJob> allTableJobMap, Analyze
+                        table.getName(), tableUpdateTime, statisticsUpdateTime);
+                return;
+            }
+
+            long timeInterval = job.getProperties().get(StatsConstants.STATISTIC_AUTO_COLLECT_INTERVAL) != null ?
+                    Long.parseLong(job.getProperties().get(StatsConstants.STATISTIC_AUTO_COLLECT_INTERVAL)) :
+                    Config.statistic_auto_collect_interval;
+
+            if (statisticsUpdateTime.plusSeconds(timeInterval).isAfter(LocalDateTime.now())) {
+                LOG.debug("statistics job doesn't work on the interval table: {}, " +
+                                "last collect time: {}, interval: {}",
+                        table.getName(), tableUpdateTime, timeInterval);
                 return;
             }
 
             double statisticAutoCollectRatio =
                     job.getProperties().get(StatsConstants.STATISTIC_AUTO_COLLECT_RATIO) != null ?
                             Double.parseDouble(job.getProperties().get(StatsConstants.STATISTIC_AUTO_COLLECT_RATIO)) :
                             Config.statistic_auto_collect_ratio;
-            if (basicStatsMeta.getHealthy() > statisticAutoCollectRatio) {
+
+            double healthy = basicStatsMeta.getHealthy();
+            if (healthy > statisticAutoCollectRatio) {
+                LOG.debug("statistics job doesn't work on health table: {}, healthy: {}, collect healthy limit: <{}",
+                        table.getName(), healthy, statisticAutoCollectRatio);
                 return;
+            } else if (healthy < Config.statistic_auto_sample_ratio) {
+                long sumDataSize = 0;
+                for (Partition partition : table.getPartitions()) {
+                    LocalDateTime partitionUpdateTime = StatisticUtils.getPartitionLastUpdateTime(partition);
+                    if (statisticsUpdateTime.isBefore(partitionUpdateTime)) {
+                        sumDataSize += partition.getDataSize();
+                    }
+                }
+
+                if (sumDataSize > Config.statistic_auto_sample_data_size) {
+                    LOG.debug("statistics job choose sample on real-time update table: {}" +
+                                    ", last collect time: {}, current healthy: {}, full collect healthy limit: {}<, " +
+                                    ", update data size: {}MB, full collect healthy data size limit: <{}MB",
+                            table.getName(), statisticsUpdateTime, healthy,
+                            Config.statistic_auto_sample_ratio, ByteSizeUnit.BYTES.toMB(sumDataSize),
+                            ByteSizeUnit.BYTES.toMB(Config.statistic_auto_sample_data_size));
+                    allTableJobMap.add(buildStatisticsCollectJob(db, table, null, columns,
+                            StatsConstants.AnalyzeType.SAMPLE, job.getScheduleType(), job.getProperties()));
+                    return;
+                }
```

Comment:
Is this logic used to auto update the type of auto collect job?  Maybe add some comments or extract this to a new method?